tools/conversion/convert_to_tflite.py

```diff
@@ -0,0 +1,1185 @@
+              'frames to save: ', new_frames_to_save,
+              'tstride: ', tstride)
+        # create (optional) expansion pointwise convolution layer
+
+        input_indexes = []
+        for i in range(num_convs):
+            input_indexes.append(len(all_layers) - frames + (i * tstride))
+
+        if xratio != 1:
+            print('---------- Insert channel multiplier pointwise conv -------------')
+            # attach output ports to inputs we will need next pass if tsize>1
+            for f in range(new_frames_to_save):
+                out_index.append(len(all_layers) - frames + f)
+                out_names.append(module_name + '_save_' + str(f))
+
+            # create input ports for required old frames if tsize>1
+            for f in range(old_frames_to_read):
+                h_name = module_name + '_history_' + str(f)
+                all_layers.append(
+                    Input(shape=(image_size[0], image_size[1], input_channels), name=h_name))
+                in_names.append(h_name)
+                in_index.append(len(all_layers) - 1)
+
+            # get weights
+            n = module_name + '.conv.' + str(s) + '.0.'
+            if n + 'weight' in weights_full:
+                weights_pt = weights_full[n + 'weight']
+                print('checkpoint: ', weights_pt.shape, )
+                weights_k = np.transpose(weights_pt, [2, 3, 1, 0])
+                bias = weights_full[n + 'bias']
+            else:
+                print('missing weight ', n + 'weight')
+                weights_k = np.random.rand(1, 1, tsize * input_channels, x_channels)
+                bias = np.zeros(x_channels)
+                fake_weights = True
+
+            expected_weights_shape = (1, 1, tsize * input_channels, x_channels)
+            print('weight shape, expected : ', expected_weights_shape,
+                  'transposed: ', weights_k.shape)
+
+            if (weights_k.shape != expected_weights_shape):
+                print('weight matrix shape is wrong, making a fake one')
+                weights_k = np.random.rand(1, 1, tsize * input_channels, x_channels)
+                bias = np.zeros(x_channels)
+                fake_weights = True
+
+            weights = [weights_k, bias]
+
+            inputs = []
+            outputs = []
+
+            for f in range(inputs_needed):
+                #                print('input index: ', len(all_layers) - inputs_needed + f, ' shape: ',
+                #                      all_layers[len(all_layers) - inputs_needed + f].shape)
+                inputs.append(all_layers[len(all_layers) - inputs_needed + f])
+                if merge_in > 0:
+                    #                    print('merge input index: ', len(all_layers) - inputs_needed + f, ' shape: ',
+                    #                          all_layers[len(all_layers) - (2 * inputs_needed) + f].shape)
+                    inputs.append(all_layers[len(all_layers) - (2 * inputs_needed) + f])
```

Comment:
This script is so messy :( Will we ever clean it up?

Reply:
Not planned in this PR.